CogniteSdk.Types/Beta/StreamRecords/Stream.cs

```diff
@@ -15,6 +17,73 @@ public class StreamWrite
         /// Stream id cannot be "logs" or "records". Max length is 100 characters.
         /// </summary>
         public string ExternalId { get; set; }
+
+        /// <summary>
+        /// Settings for the stream. Required field.
+        /// </summary>
+        [JsonPropertyName("settings")]
+        public StreamSettings Settings { get; set; }
+    }
+
+    /// <summary>
+    /// Settings for a stream.
+    /// </summary>
+    public class StreamSettings
+    {
+        /// <summary>
+        /// Template settings for the stream.
+        /// </summary>
+        [JsonPropertyName("template")]
+        public StreamTemplateSettings Template { get; set; }
+    }
+
+    /// <summary>
+    /// The name of the stream template.
+    /// </summary>
+    [JsonConverter(typeof(StreamTemplateNameConverter))]
+    public enum StreamTemplateName
+    {
+        /// <summary>
+        /// Immutable test stream template
+        /// </summary>
+        ImmutableTestStream,
+
+        /// <summary>
+        /// Immutable data staging template
+        /// </summary>
+        ImmutableDataStaging,
+
+        /// <summary>
+        /// Immutable normalized data template
+        /// </summary>
+        ImmutableNormalizedData,
+
+        /// <summary>
+        /// Immutable archive template
```

Comment:
you don't need custom converter for this.
just use `[JsonConverter(typeof(JsonStringEnumConverter))]`, look at `CogniteSdk.Types/Alpha/Simulators/SimulatorModelRevision.cs`